### PR TITLE
Change the minimum version for using autoscaling/v2 api to 1.23

### PR DESCRIPTION
**Pull request by jlawrienyt**
This addresses issue #3154

### Checklist

- [x] Changelog updated or skip changelog label added

### CODE REVIEW

File: .changelog/3155.fixed.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+Change the minimum version for using autoscaling/v2 api to 1.23
```
Comment (sumo-drosiek) [inline .changelog/3155.fixed.txt:1]:
```suggestion
feat(helm): change the minimum version for using autoscaling/v2 api to 1.23
```

**Comment by swiatekm-sumo** [comment]
From the issue, it sounds like you run into trouble if you have the Chart installed on a lower version (and therefore are using the deprecated api group), and then upgrade to 1.26, where that api group was removed. Kubernetes itself has no issue with this, and in fact will internally convert to the latest supported api group, but Helm itself won't, and reports an issue with the existing manifests - even if the upgrade would, in fact, fix the problem.

**Comment by sumo-drosiek** [comment]
> From the issue, it sounds like you run into trouble if you have the Chart installed on a lower version (and therefore are using the deprecated api group), and then upgrade to 1.26, where that api group was removed. Kubernetes itself has no issue with this, and in fact will internally convert to the latest supported api group, but Helm itself won't, and reports an issue with the existing manifests - even if the upgrade would, in fact, fix the problem.

Then it's better to check for `1.23`, so the issue won't be a problem if he first installation has been done on at least `1.23`

**Comment by swiatekm-sumo** [comment]
> 



> > From the issue, it sounds like you run into trouble if you have the Chart installed on a lower version (and therefore are using the deprecated api group), and then upgrade to 1.26, where that api group was removed. Kubernetes itself has no issue with this, and in fact will internally convert to the latest supported api group, but Helm itself won't, and reports an issue with the existing manifests - even if the upgrade would, in fact, fix the problem.
> 
> Then it's better to check for `1.23`, so the issue won't be a problem if he first installation has been done on at least `1.23`

That's what this PR does, right?

**Review by sumo-drosiek — APPROVED** [review]

**Pull request merged**